### PR TITLE
Fix Dockerfile

### PR DESCRIPTION
**Pull request by Luzifer**
- [x] Use `golang:1.4` as suggested in [install documentation](http://ipfs.io/docs/install/)
- [x] Add port `8080/tcp` to port command for HTTP access to IPFS
- [x] Figure out how to let the daemon listen on `0.0.0.0` without additional script. (Otherwise the port access will not work as the daemon is listening on `127.0.0.1` by default)

### CODE REVIEW

**Comment by jbenet** [comment]
> Figure out how to let the daemon listen on `0.0.0.0`

Ah yes, this change in the default broke the dockerfile. We didnt notice it because we dont have tests for that. We could do this multiple ways, from easiest to hardest:

- make the dckerfile use `ipfs config` to change config values
- add a routine into the ipfs daemon command that runs a different set of defaults saner for `ipfs daemon --init`
- fix #251 (and maybe #387)

**Comment by Luzifer** [comment]
Change works as expected:

```
luzifer@knut-workstation01 ~/g/s/g/L/go-ipfs (fix_dockerfile+)> docker build .
Sending build context to Docker daemon 19.99 MB
Sending build context to Docker daemon
Step 0 : FROM golang:1.4
511136ea3c5a: Already exists
8771fbfe935c: Already exists
0e30e84e9513: Already exists
c90a56bfe7dd: Already exists
6b030fdd4748: Already exists
5b691e49c664: Already exists
bced94a2bb11: Already exists
dcb230f60b2a: Already exists
b172f9f42707: Already exists
a495469fdb31: Already exists
78ab42d1942d: Already exists
58e51ffff67e: Already exists
03049bfbf1be: Already exists
d963de8784a3: Already exists
73a01651fc2c: Already exists
golang:1.4: The image you are pulling has been verified. Important: image verification is a tech preview feature and should not be relied on to provide security.
Status: Image is up to date for golang:1.4
 ---> 73a01651fc2c
Step 1 : MAINTAINER Brian Tiger Chow <btc@perfmode.com>
 ---> Running in 53a6df74d72a
 ---> a3df043a7fcc
Removing intermediate container 53a6df74d72a
Step 2 : ADD . /go/src/github.com/jbenet/go-ipfs
 ---> 4900ae898fd7
Removing intermediate container 764053d1822e
Step 3 : RUN cd /go/src/github.com/jbenet/go-ipfs/cmd/ipfs && go install
 ---> Running in 2886c5a7d3f7
 ---> 8582db6e9e64
Removing intermediate container 2886c5a7d3f7
Step 4 : RUN echo -n "#!/bin/bash\nipfs init\nipfs config Addresses.API /ip4/0.0.0.0/tcp/5001\nipfs config Addresses.Gateway /ip4/0.0.0.0/tcp/8080\nipfs daemon" > /usr/local/bin/start_ipfs &&     chmod 755 /usr/local/bin/start_ipfs
 ---> Running in 556dcfcdf53e
 ---> 5a981854376a
Removing intermediate container 556dcfcdf53e
Step 5 : EXPOSE 4001 5001 4002/udp 8080
 ---> Running in 7481ba35637f
 ---> 92fc89560ccd
Removing intermediate container 7481ba35637f
Step 6 : ENTRYPOINT /usr/local/bin/start_ipfs
 ---> Running in 388af0981f07
 ---> 9b0aafed4711
Removing intermediate container 388af0981f07
Successfully built 9b0aafed4711
luzifer@knut-workstation01 ~/g/s/g/L/go-ipfs (fix_dockerfile+)> docker run -ti 9b0aafed4711
initializing ipfs node at /root/.go-ipfs
generating 4096-bit RSA keypair...done
peer identity: QmTs2qPMp2R9ndRjyjCCnJ9zT4u2xcQVE1Sasp8X9p4MK5
to get started, enter:

	ipfs cat /ipfs/QmV9E5oNhFDWeRV8NQ91G7vp3pvff2S4UYkn89T5koSzqo/readme

Initializing daemon...
API server listening on /ip4/0.0.0.0/tcp/5001
Gateway server listening on /ip4/0.0.0.0/tcp/8080
^CReceived interrupt signal, shutting down...
luzifer@knut-workstation01 ~/g/s/g/L/go-ipfs (fix_dockerfile+)>
```

@jbenet PTAL

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,17 @@
-FROM golang:1.3
+FROM golang:1.4
 MAINTAINER Brian Tiger Chow <btc@perfmode.com>
 
 ADD . /go/src/github.com/jbenet/go-ipfs
 RUN cd /go/src/github.com/jbenet/go-ipfs/cmd/ipfs && go install
 
-EXPOSE 4001 5001 4002/udp
+RUN echo -n "#!/bin/bash\nipfs init\nipfs config Addresses.API /ip4/0.0.0.0/tcp/5001\nipfs config Addresses.Gateway /ip4/0.0.0.0/tcp/8080\nipfs daemon" > /usr/local/bin/start_ipfs && \
```
Comment (jbenet) [inline Dockerfile]: maybe we could put this into a script in `bin/` (maybe `bin/container_daemon` or something) for cleaner readability.

**Comment by jbenet** [comment]
this approach LGTM. if @briantigerchow wants to take a look, his input would be good. One minor change above (moving the sh script into its own file) but otherwise LGTM!

**Comment by jbenet** [comment]
LGTM, thanks @Luzifer!

**Pull request merged**